model/bookImplementation.go

```diff
@@ -0,0 +1,91 @@
+package model
+
+import (
+	"cloud.google.com/go/datastore"
+	"context"
+	"errors"
+	"fmt"
+	"google.golang.org/api/iterator"
+	"math/rand/v2"
+)
+
+type BookInterface interface {
+	GetByKey(id int) (Book, error)
+	ListAll(query *datastore.Query) []Book
```

Comment:
Keep in mind separation of concerns. If the API layer is doing datastore stuff, then it's a bit of a code smell that the impl is leaking out of the interface. I would have passed all the list filters as individual params, and have the model construct the query itself